reolink_aio/api.py

```diff
@@ -385,6 +385,26 @@ def is_admin(self) -> bool:
         """
         return self.user_level == "admin"
 
+    def get_timezone(self) -> tzinfo|None:
+        """Get the timezone of the device"""
+        if self._time_settings is None:
+            return None
+        return Reolink_timezone.get(self._time_settings)
+
+    def get_time(self) -> datetime|None:
+        """Get the approximate "current" time of the camera"""
+        if self._time_settings is None:
+            return None
+        return datetime.now().astimezone(self.get_timezone()) + timedelta(seconds=self._host_time_difference)
+
+    async def async_get_time(self) -> datetime:
+        """Get the current time of the camera"""
+        await self.get_state("GetTime")
```

Comment:
Please only retrieve the GetTime again if it is not already available.
So move this in the `if self._time_settings is None:`
This is normally already retrieved in the `get_host_data` call.

Reply:
As a follow up, the proper usage would be to call get_time and if it returns None (thus no time data available) call async_get_time to ensure a time or throw, otherwise get_time should be good enough for most situations. (especially since the time data is pretty much "stale" after it is retrieved since it static.)

Reply:
I just though of a question, are you ok with get_* or should this follow the more "pythonic" approach of just a timezone() method and datetime()/async_datetime() methods? I feel leaving them methods signifies that there is work done behinds the scenes/readonly, whereas properties would denote "instant" access to a stored value.

Reply:
Actually if no IO is done I would not start with get_*, but I don't have strong fealings about this.
If IO is done I would start with get.